pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -117,11 +117,23 @@ func (s *SpdyRoundTripper) dial(req *http.Request) (net.Conn, error) {
 
 	tlsConn := tls.Client(rwc, s.tlsConfig)
 
+	// Return if we were configured to skip validation
+	if s.tlsConfig != nil && s.tlsConfig.InsecureSkipVerify {
+		return tlsConn, nil
+	}
+
 	// need to manually call Handshake() so we can call VerifyHostname() below
 	if err := tlsConn.Handshake(); err != nil {
 		return nil, err
 	}
 
+	state := tlsConn.ConnectionState()
+	for _, peerCert := range state.PeerCertificates {
+		if err := peerCert.VerifyHostname(host); err != nil {
```

Comment:
I think just the InsecureSkipVerify is needed... from what I can see, in go1.5 it still sets verifiedChains during the handshake. Can you try without this change?